controllers/chat/ChatController.js

```diff
@@ -426,6 +426,135 @@ module.exports = {
+
+      await channel.updatePartial({
+        set: {
+          channel_type: CHANNEL_TYPE.ANONYMOUS,
+          anon_user_info_color_code,
+          anon_user_info_color_name,
+          anon_user_info_emoji_code,
+          anon_user_info_emoji_name
+        }
+      });
+
+      // Default mock chat in case of channel is blocked
+      let chat = {
+        message: {
+          id: uuidv4(),
+          cid: createdChannel?.channel?.id,
+          text: message,
+          message,
+          user: {
+            id: req.userId,
+            name: `Anonymous ${anon_user_info_emoji_name}`,
+            image: '',
+            username: `Anonymous ${anon_user_info_emoji_name}`
+          },
+          created_at: new Date(),
+          updated_at: new Date()
+        }
+      };
+
+      if (!createdChannel?.channel?.is_channel_blocked) {
+        chat = await channel.sendMessage({
+          user_id: req.userId,
+          text: message,
+          ...req.body
+        });
+      }
+
+      const targets = members.filter((member) => member !== req.userId);
+      await targets.map(async (target) => {
+        const exist = await ChatAnonUserInfo.count({
+          where: {
+            channel_id: channel.id,
+            my_anon_user_id: req.userId,
+            target_user_id: target
+          }
+        });
+
+        if (!exist) {
+          await ChatAnonUserInfo.create({
+            channel_id: channel.id,
+            my_anon_user_id: req.userId,
+            target_user_id: target,
+            anon_user_info_color_code,
+            anon_user_info_color_name,
+            anon_user_info_emoji_code,
+            anon_user_info_emoji_name
+          });
+        }
+      });
+
+      const targetsUserModel = await UsersFunction.findMultipleUsersById(User, targets);
+      targetsUserModel.push({
+        user_id: req.userId,
+        username: `Anonymous ${anon_user_info_emoji_name}`,
+        profile_pic_path: '',
+        anon_user_info_color_code,
+        anon_user_info_color_name,
+        anon_user_info_emoji_code,
+        anon_user_info_emoji_name
+      });
+
+      // get 100 messages
+      const channelFilters = {cid: 'messaging:' + channel.id};
+      const messageFilters = {created_at: {$lte: new Date()}};
+      const messageHistory = await client.search(channelFilters, messageFilters, {
+        sort: [{updated_at: -1}],
+        limit: 100
+      });
+
+      await client.disconnectUser();
+
+      const response = {
+        ...chat,
+        members: targetsUserModel,
+        messageHistory: messageHistory.results
+      };
+
+      if (createdChannel?.channel?.is_channel_blocked) {
+        return res.status(400).json(responseError('Channel is blocked', response));
+      }
+
+      return res.status(200).json(responseSuccess('sent', response));
+    } catch (error) {
+      await client.disconnectUser();
+      return ErrorResponse.e400(res, error.message);
+    }
+  },
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `initChatMoveToAnon` function is quite large and does a lot of different things. This makes it hard to understand, maintain, and test. It would be better to break this function down into smaller, more manageable functions. For example, you could have separate functions for creating the user, creating the channel, sending the message, saving anonymous user information, and retrieving message history.

Also, there's no error handling for the case when `StreamChat.getInstance()` fails. If it fails, the rest of the code will likely throw an error because it assumes that `client` is a valid instance of `StreamChat`.

```diff
-    const client = StreamChat.getInstance(process.env.API_KEY, process.env.SECRET);
+    let client;
+    try {
+      client = StreamChat.getInstance(process.env.API_KEY, process.env.SECRET);
+    } catch (error) {
+      return ErrorResponse.e400(res, 'Failed to get StreamChat instance');
+    }
```

Lastly, the use of `Array.prototype.map` at line 499 is incorrect. The `map` function is used to transform each element of an array and create a new array with the transformed elements. However, in this case, you're using it for its side effect (creating `ChatAnonUserInfo` instances), not for transforming elements. You should use `Array.prototype.forEach` or a `for...of` loop instead.

```diff
-      await targets.map(async (target) => {
+      for (const target of targets) {
```
And don't forget to add `await` before `ChatAnonUserInfo.create` inside the loop.

```diff
-            await ChatAnonUserInfo.create({
+            await ChatAnonUserInfo.create({
```
Then close the loop after the creation of `ChatAnonUserInfo`.

```diff
+      }
```
This will ensure that all `ChatAnonUserInfo` instances are created before the function continues.


<!-- This is an auto-generated comment by OSS CodeRabbit -->